core/src/main/java/org/testcontainers/containers/traits/Trait.java

```diff
@@ -0,0 +1,10 @@
+package org.testcontainers.containers.traits;
+
+import com.github.dockerjava.api.command.CreateContainerCmd;
+import org.testcontainers.containers.TestContainer;
+
+public interface Trait<T extends TestContainer<T>> {
+
+    void configure(T container, CreateContainerCmd createContainerCmd);
```

Comment:
we might want to add more methods here in future, i.e. lifecycle (starting/started)